msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/CurveTool.java

```diff
@@ -28,117 +28,125 @@
-	 * @param scanNumber a {@link java.lang.Integer} object. This is random scan numbers for which we're finding gaussian fit.
-	 * </p>
-	 */
-	private void createDataFile(List<SliceSparseMatrix.VerticalSliceDataPoint> datapoint,double[] parameters,int scanNumber){
-		double gaussianY [] = new double[datapoint.size()];
-		String filename = "output"+ scanNumber +".txt";
-		try{
-			File outputFile = new File(TEST_DATA_PATH + filename);
-			StringBuffer buffer = new StringBuffer ();
-			FileWriter writer = new FileWriter(outputFile);
-			
-			for(int i=0;i<datapoint.size();i++){
-				gaussianY[i] = parameters[0]*Math.exp(-Math.pow((datapoint.get(i).mz-parameters[1]), 2)/(2*Math.pow(parameters[2],2)));
-				buffer.append(datapoint.get(i).mz).append(",").append(datapoint.get(i).intensity).append(",").append(gaussianY[i]).append("\r\n");
-				writer.write(buffer.toString());
-				buffer = new StringBuffer();
-			}
-			writer.close();
-		}
-		
-		catch(Exception e){
-			
-		}
-		
-	}
+
+  private static final String TEST_DATA_PATH = "src/test/resources/";
+
+  private SliceSparseMatrix objSliceSparseMatrix;
+
+  /**
+   * <p>
+   * CurveTool constructor takes object of SliceSparseMatrix class.
+   * </p>
+   */
+  public CurveTool(SliceSparseMatrix sliceSparseMatrix) {
+    objSliceSparseMatrix = sliceSparseMatrix;
+  }
+
+  /**
+   * <p>
+   * estimateFwhmMs method estimates the FWHM for given number of random scans.
+   * 
+   * @param numberOfScansForFWHMCalc a {@link java.lang.Integer} object. This is number scans
+   *        required for estimation of fwhm.
+   * 
+   * @return fwhm a {@link java.lang.Double} object.This is Full width half maximum.
+   *         </p>
+   */
+  public double estimateFwhmMs(int numberOfScansForFWHMCalc) {
+
+    double sigma = 0;
+    int countProperIteration = 0;
+    int countTotalIteration = 0;
+
+    while (countProperIteration < numberOfScansForFWHMCalc) {
+      countTotalIteration++;
+      int matrixIndex = 0;
+
+
+      if (countTotalIteration > objSliceSparseMatrix.getSizeOfRawDataFile()) {
+        System.out.println(countTotalIteration);
+        throw new IllegalArgumentException("Cannot calculate FWHM.");
+      }
+
+      Random generator = new Random();
+      int randInt = generator.nextInt(objSliceSparseMatrix.getSizeOfRawDataFile());
+      List<SliceSparseMatrix.VerticalSliceDataPoint> verticalSlice =
+          objSliceSparseMatrix.getVerticalSlice(randInt);
+
+      if (verticalSlice == null)
+        continue;
+
+      WeightedObservedPoints obs = new WeightedObservedPoints();
+
+      for (SliceSparseMatrix.VerticalSliceDataPoint datapoint : verticalSlice) {
+        obs.add(datapoint.mz, datapoint.intensity);
+        matrixIndex++;
+      }
+
+      try {
+        double[] parameters = GaussianCurveFitter.create().fit(obs.toList());
+        // createDataFile(verticalSlice,parameters,randInt);
+        sigma += 2.35482 * parameters[2];
+
+      } catch (Exception e) {
+        continue;
```

Comment:
I also think you should catch specific exceptions and handling each appropriately.